xknx/devices/climate_mode.py

```diff
@@ -180,103 +188,49 @@ def operation_modes(self):
             return []
         return self.operation_modes_
 
-    def guess_operation_modes(self):
-        """Guess operation modes from group addresses."""
-        # All operation modes supported
-        if self.group_address_operation_mode is not None:
-            return [HVACOperationMode.AUTO, HVACOperationMode.COMFORT,
-                    HVACOperationMode.STANDBY, HVACOperationMode.NIGHT,
-                    HVACOperationMode.FROST_PROTECTION]
-        if self.group_address_controller_status is not None:
-            return [HVACOperationMode.COMFORT, HVACOperationMode.STANDBY,
-                    HVACOperationMode.NIGHT, HVACOperationMode.FROST_PROTECTION]
-        if self.group_address_controller_mode is not None:
-            return [HVACOperationMode.STANDBY, HVACOperationMode.AUTO, HVACOperationMode.HEAT,
-                    HVACOperationMode.COOL, HVACOperationMode.FAN_ONLY, HVACOperationMode.DRY]
-
-        # Operation modes only supported partially
+    def gather_operation_modes(self):
+        """Gather operation modes from RemoteValues."""
         operation_modes = []
-        if self.group_address_operation_mode_comfort:
-            operation_modes.append(HVACOperationMode.COMFORT)
-        operation_modes.append(HVACOperationMode.STANDBY)
-        if self.group_address_operation_mode_night:
-            operation_modes.append(HVACOperationMode.NIGHT)
-        if self.group_address_operation_mode_protection:
-            operation_modes.append(HVACOperationMode.FROST_PROTECTION)
-        return operation_modes
+        for rv in self.__iter_remote_values():
+            if rv.writable:
+                operation_modes.extend(rv.supported_operation_modes())
+        # remove duplicates
+        return list(set(operation_modes))
```

Comment:
Don't really like this, but I haven't found a more elegant way.

Reply:
This should also work and would keep the order:
`operation_modes = list(dict.fromkeys(operation_modes))`